reflex/components/core/banner.py

```diff
@@ -19,50 +18,45 @@
 from reflex.components.sonner.toast import Toaster, ToastProps
 from reflex.constants import Dirs, Hooks, Imports
 from reflex.constants.compiler import CompileVars
+from reflex.ivars.base import ImmutableVar, LiteralVar
+from reflex.ivars.function import FunctionStringVar
+from reflex.ivars.number import BooleanVar
+from reflex.ivars.sequence import LiteralArrayVar
 from reflex.utils.imports import ImportDict, ImportVar
-from reflex.utils.serializers import serialize
-from reflex.vars import Var, VarData
+from reflex.vars import ImmutableVarData, Var, VarData
 
 connect_error_var_data: VarData = VarData(  # type: ignore
     imports=Imports.EVENTS,
     hooks={Hooks.EVENTS: None},
 )
 
-connect_errors: Var = Var.create_safe(
+connect_errors: Var = ImmutableVar.create_safe(
     value=CompileVars.CONNECT_ERROR,
-    _var_is_local=True,
-    _var_is_string=False,
     _var_data=connect_error_var_data,
 )
 
-connection_error: Var = Var.create_safe(
-    value="(connectErrors.length > 0) ? connectErrors[connectErrors.length - 1].message : ''",
-    _var_is_local=False,
-    _var_is_string=False,
+connection_error: Var = ImmutableVar.create_safe(
+    value="((connectErrors.length > 0) ? connectErrors[connectErrors.length - 1].message : '')",
     _var_data=connect_error_var_data,
 )
 
-connection_errors_count: Var = Var.create_safe(
+connection_errors_count: Var = ImmutableVar.create_safe(
     value="connectErrors.length",
-    _var_is_string=False,
-    _var_is_local=False,
     _var_data=connect_error_var_data,
 )
 
-has_connection_errors: Var = Var.create_safe(
-    value="connectErrors.length > 0",
-    _var_is_string=False,
+has_connection_errors: Var = ImmutableVar.create_safe(
+    value="(connectErrors.length > 0)",
     _var_data=connect_error_var_data,
-).to(bool)
+).to(BooleanVar)
 
-has_too_many_connection_errors: Var = Var.create_safe(
-    value="connectErrors.length >= 2",
-    _var_is_string=False,
+has_too_many_connection_errors: Var = ImmutableVar.create_safe(
+    value="(connectErrors.length >= 2)",
     _var_data=connect_error_var_data,
-).to(bool)
+).to(BooleanVar)
 
 
-class WebsocketTargetURL(Bare):
+class WebsocketTargetURL:
```

Comment:
can probably be a subclass of ImmutableVar now

and `add_imports` can be removed in favor of the included VarData with the necessary imports